File: event-core/src/main/java/com/event/model/response/EventResponse.java
package com.event.model.response;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;

@JsonInclude(JsonInclude.Include.ALWAYS)
public record EventResponse(
    Long contentId,
    String title,
    LocalDateTime createdTime,
    LocalDateTime modifiedTime,
    String addr1,
    String addr2,
    String area,
    String firstImage,
    String firstImage2,
    Double mapX,
    Double mapY,
    String zipCode,
    String homepage,
    String overview,
    LocalDate eventStartDate,
    LocalDate eventEndDate,
    String playTime,
    String useTimeFestival,
    String sponsor1,
    String sponsor1Tel,
    String sponsor2,
    String sponsor2Tel,
    Instant dbUpsertedAt,
    boolean favoriteStatus
) {
}

File: frontend/src/app/_types/responses/event-response.ts
export interface EventResponse {
  contentId: number;
  title: string;
  createdTime: string;
  modifiedTime: string;
  addr1: string;
  addr2: string;
  area: string;
  firstImage: string;
  firstImage2: string;
  mapX: number;
  mapY: number;
  zipCode: string;
  homepage: string;
  overview: string;
  eventStartDate: string;
  eventEndDate: string;
  playTime: string;
  useTimeFestival: string;
  sponsor1: string;
  sponsor1Tel: string;
  sponsor2: string;
  sponsor2Tel: string;
  dbUpdatedAt: string;
}

export interface ErrorResponse { 
  message: string;
};
